chain-tx-enclave-next/mls/src/group.rs

```diff
@@ -262,103 +358,47 @@ impl GroupAux {
 
-        let adds = commit_content
-            .adds
-            .iter()
-            .map(|proposal_id| {
-                add_proposals_ids
-                    .get(proposal_id)
-                    .cloned()
-                    .cloned()
-                    .ok_or(ProcessCommitError::CommitError)
-            })
-            .collect::<Result<Vec<_>, _>>()?;
-        // FIXME: updates
-        // FIXME: removes
         let mut updated_tree = self.tree.clone();
-
-        let _ = updated_tree.update(&adds, &[], &[]);
-        // "Verify that the path value is populated if either of the updates or removes vectors has length greater than zero
-        // all of the updates, removes, and adds vectors are empty."
-        if commit_content.path.is_none()
-            && ((!commit_content.updates.is_empty() || !commit_content.removes.is_empty())
-                || (commit_content.adds.is_empty()
-                    && commit_content.updates.is_empty()
-                    && commit_content.removes.is_empty()))
-        {
-            return Err(ProcessCommitError::CommitError);
-        }
-
-        let updated_epoch = self.context.epoch + 1;
-        let mut updated_group_context = self.context.clone();
-        updated_group_context.epoch = updated_epoch;
-        // "If the path value is populated: Process the path value..."
-
-        // apply commit path, leaf keypackage and path secrets
-        let commit_secret = if let Some(path) = &commit_content.path {
-            let sender = LeafSize(commit.content.sender.sender as usize);
-            let (leaf_parent_hash, root_secret) = updated_tree.apply_path_secrets(
-                sender,
-                &self.context.get_encoding(),
-                &path.nodes,
-                &self.owned_kp.init_private_key,
-            )?;
-            // Verify that the KeyPackage has a `parent_hash` extension and that its value
-            // matches the new parent of the sender's leaf node.
-            let ext = path
-                .leaf_key_package
-                .payload
-                .find_extension::<ext::ParentHashExt>()?;
-            if !bool::from(ext.0.ct_eq(&leaf_parent_hash)) {
-                return Err(ProcessCommitError::LeafParentHashDontMatch);
-            }
-            updated_tree.set_package(sender, path.leaf_key_package.clone());
-            root_secret
-        } else {
-            // "If the path value is not populated: ..."
-            SecretVec::new(vec![0; self.tree.cs.hash_len()])
-        };
-        updated_group_context.tree_hash = updated_tree.compute_tree_hash();
+        let (_, commit_secret) = updated_tree.apply_commit(
+            &self.context.get_encoding(),
+            &self.owned_kp.init_private_key,
+            &commit_content,
+        )?;
+        // "If the path value is not populated: ..."
+        let commit_secret =
+            commit_secret.unwrap_or_else(|| SecretVec::new(vec![0; self.tree.cs.hash_len()]));
```

Comment:
should it also validate that for those two cases (`pathRequired = haveUpdateOrRemove || haveNoProposalsAtAll`), it really indeed populated the path instead of continuing with empty vec?

Reply:
> should it also validate that for those two cases (`pathRequired = haveUpdateOrRemove || haveNoProposalsAtAll`), it really indeed populated the path instead of continuing with empty vec?

done